crates/proof-of-sql/src/sql/postprocessing/evaluate.rs

```diff
@@ -0,0 +1,267 @@
+                            })
+                            .collect();
+                        Ok(OwnedColumn::Boolean(res))
+                    }
+                    BinaryOperator::LessThanOrEqual => {
+                        let res: Vec<bool> = (0..length)
+                            .map(|i| {
+                                let lhs = left.scalar_at(i).unwrap() * lhs_upscale_factor;
+                                let rhs = right.scalar_at(i).unwrap() * rhs_upscale_factor;
+                                lhs.signed_cmp(&rhs) != Ordering::Greater
+                            })
+                            .collect();
+                        Ok(OwnedColumn::Boolean(res))
+                    }
+                    _ => unreachable!(),
+                }
+            }
+            BinaryOperator::Add => Ok(add_subtract_owned_columns(&left, &right, false)),
+            BinaryOperator::Subtract => Ok(add_subtract_owned_columns(&left, &right, true)),
+            BinaryOperator::Multiply => Ok(multiply_owned_columns(&left, &right)),
+            BinaryOperator::Division => try_divide_owned_columns(&left, &right),
+        }
+    }
+}
+
+/// If the two column types are compatible for a binary operation, return the result type.
+/// Otherwise, return None.
+pub(crate) fn get_binary_operation_result_type(
+    left_dtype: &ColumnType,
+    right_dtype: &ColumnType,
+    binary_operator: BinaryOperator,
+) -> Option<ColumnType> {
+    match binary_operator {
+        BinaryOperator::And | BinaryOperator::Or => matches!(
+            (left_dtype, right_dtype),
+            (ColumnType::Boolean, ColumnType::Boolean)
+        )
+        .then_some(ColumnType::Boolean),
+        BinaryOperator::Equal => (matches!(
+            (left_dtype, right_dtype),
+            (ColumnType::VarChar, ColumnType::VarChar)
+                | (ColumnType::TimestampTZ(_, _), ColumnType::TimestampTZ(_, _))
+                | (ColumnType::Boolean, ColumnType::Boolean)
+                | (_, ColumnType::Scalar)
+                | (ColumnType::Scalar, _)
+        ) || (left_dtype.is_numeric() && right_dtype.is_numeric()))
+        .then_some(ColumnType::Boolean),
+        BinaryOperator::GreaterThanOrEqual | BinaryOperator::LessThanOrEqual => {
+            if left_dtype == &ColumnType::VarChar || right_dtype == &ColumnType::VarChar {
+                return None;
+            }
+            // Due to constraints in bitwise_verification we limit the precision of decimal types to 38
+            if let ColumnType::Decimal75(precision, _) = left_dtype {
+                if precision.value() > 38 {
+                    return None;
+                }
+            }
+            if let ColumnType::Decimal75(precision, _) = right_dtype {
+                if precision.value() > 38 {
+                    return None;
+                }
+            }
+            // TODO: inequality support for timestamps
```

Comment:
timestamp inequality support was merged in #52, is this comment referring to something different?